plugins/source/gandi/codegen/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/cloudquery/cloudquery/plugins/source/gandi/codegen/recipes"
+	"github.com/cloudquery/plugin-sdk/codegen"
+)
+
+//go:embed templates/*.go.tpl
+var templatesFS embed.FS
+
+func main() {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		log.Fatal("Failed to get caller information")
+	}
+	codegenDir := path.Join(path.Dir(filename), "..", "resources", "services")
+
+	var resources []recipes.Resource
+	resources = append(resources, recipes.CertificateResources()...)
+	resources = append(resources, recipes.DomainResources()...)
+	resources = append(resources, recipes.LiveDNSResources()...)
+	resources = append(resources, recipes.SimpleHostingResources()...)
+
+	for _, r := range resources {
+		generateTable(codegenDir, r)
+	}
+}
+
+func generateTable(basedir string, r recipes.Resource) {
+	var err error
+
+	log.Println("Generating table", r.TableName)
+	opts := []codegen.TableOption{
+		codegen.WithSkipFields(r.SkipFields),
+		codegen.WithExtraColumns(r.ExtraColumns),
+		codegen.WithPKColumns(r.PKColumns...),
+	}
+	if r.UnwrapEmbeddedStructs {
+		opts = append(opts, codegen.WithUnwrapAllEmbeddedStructs())
+	}
+	r.Table, err = codegen.NewTableFromStruct(r.TableName, r.DataStruct, opts...)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	r.Table.Resolver = r.ResolverFuncName
+	r.Table.Multiplex = r.Multiplex
+	r.ImportClient = strings.HasPrefix(r.Multiplex, "client.")
+	r.Table.Relations = r.Relations
+	r.Table.PreResourceResolver = r.PreResourceResolver
+	r.Table.PostResourceResolver = r.PostResourceResolver
+
+	for _, c := range r.Table.Columns {
+		if strings.HasPrefix(c.Resolver, "client.") {
+			r.ImportClient = true
+		}
+	}
+
+	mainTemplate := r.Template + ".go.tpl"
+	tpl, err := template.New(mainTemplate).ParseFS(templatesFS, "templates/"+mainTemplate)
+	if err != nil {
+		log.Fatal(fmt.Errorf("failed to parse gandi templates: %w", err))
```

Comment:
Not blocking this PR, but is there some value in sharing this method (by parameterizing the name of the plugin) between all plugins ?

Reply:
@cqgaurav How we do things change between plugins especially with complex SDKs. It would mean lots of "flags" to understand and configure per plugin... Maybe a very limited set can be "copied" and configured per plugin (ie. this one was copied from the Cloudflare codegen and slightly modified). Eventually parts of it can be extracted as utilities and put into sdk the codegen package which provides main codegen functionality.